services/metadata_service/api/task.py

```diff
@@ -181,7 +181,11 @@ async def create_task(self, request):
             return web.Response(status=400, body=json.dumps(
                 {"message": "provided task_name may not be a numeric"}))
 
-        run_number, run_id = await self._db.get_run_ids(flow_id, run_number)
+        run = await self._db.get_run_ids(flow_id, run_number)
+        if run.response_code != 200:
+            return DBResponse(400, {"message": "need to register run_id and task_id first"})
```

Comment:
same comment as for steps. Tasks in the V3 schema also have a foreign key covering this, so consider handling the specific error instead.